src/ssl.c

```diff
@@ -13869,6 +13869,40 @@ static int wolfSSL_DupSessionEx(const WOLFSSL_SESSION* input,
     WOLFSSL_SESSION* output, int avoidSysCalls, byte* ticketNonceBuf,
     byte* ticketNonceLen, byte* preallocUsed);
 
+#ifdef WOLFSSL_DTLS_NO_HVR_ON_RESUME
+int TlsSessionInCache(const byte *id, byte *found)
```

Comment:
Is there much in common here with the code in wolfSSL_GetSesionFromCache?
If so, maybe have wolfSSL_GetSesionFromCache call a modified version of this function.
Don't want to have session lookup code in multiple places if possible.

Reply:
This makes sense, I'll try to isolate the lookup code from the code that deals with the preallocation of buffers.